dashboard/db/migrate/20220420202427_create_code_review_requests.rb

```diff
@@ -0,0 +1,18 @@
+class CreateCodeReviewRequests < ActiveRecord::Migration[6.0]
+  def change
+    create_table :code_review_requests do |t|
+      t.integer :user_id, null: false
+      t.integer :script_id, null: false
+      t.integer :level_id, null: false
+      t.integer :project_id, null: false
+      t.string :project_version, null: false
+      t.datetime :closed_at
+      t.datetime :deleted_at
+
+      t.timestamps
+
+      t.index [:user_id, :script_id, :level_id, :closed_at, :deleted_at],
```

Comment:
Currently, `closed_at` will have the same flaw as `deleted_at` -- it is possible to have multiple rows with `closed_at` equal to null.  Once we figure out what the correct pattern is for `deleted_at`, I will apply it to `closed_at` as well.

Reply:
Does `closed_at` need to be included in this index at all? I was expecting, based on the spec, a code review to be unique across a user, script, and level. It seems like `deleted_at` is there to be able to reset progress, so that also makes sense in the index. But I can't see how `closed_at` factors into this uniqueness

Reply:
My understanding from the spec (which I'm now questioning 😄) is that, for a given script+level, we want to allow the author to open and close review requests multiple times.  I think this means that we need to allow multiple code review requests for a given user, script, and level.  We also probably want to enforce that there is only one _open_ code review request for a given user, script, and level at any time.  Right?  (cc: @maureensturgeon)

Reply:
I confirmed with Maureen that we want to enforce that there is at most one non-closed, non-deleted code_review_request row per user_id + script_id + level_id.  Does this index look like it matches that intent?

Reply:
I confirmed with Maureen that we want to enforce that there is at most one non-closed, non-deleted code_review_request row per user_id + script_id + level_id.  Does this index look like it matches that intent?

Reply:
The [spec says that code reviews cannot be reopened](https://docs.google.com/document/d/1ZtvE5fZPbndZsT3gGLc1ZuH7WcdSRdCj1Tc09EwCQJ4/edit?disco=AAAAX3BAMwc), so I assumed that we would want to enforce that there was one non-deleted code_review_request per user_id + script_id + level_id. But if the goal of this index is only to enforce that one non-deleted, non-closed row exists on these axis then this index makes sense.

Reply:
Ohhhh.  I took that to mean that we can't reopen a closed code review but we could open a new one against a given user_id, script_id, level_id.  Thanks for pushing on this, let me check with Maureen.

Reply:
aha! that would make a lot of sense then and this index would look good